src/dev/nuttx/lv_nuttx_libuv.h

```diff
@@ -25,6 +25,8 @@ extern "C" {
  *      DEFINES
  *********************/
 
+#define LV_EVENT_VSYNC (_LV_EVENT_LAST + 1)
```

Comment:
Let's define it in `lv_event.h`.

Reply:
You can use `lv_event_register_id` too.

In the practice what do you exactly do in the `LV_EVENT_VSYNC` event?

Reply:
To optimize the refresh rate, it's crucial to synchronize the entire refresh process with the display's hardware vsync signal. If not, the refresh frequency driven by CPU could fall out of sync with display's frequency, eventually resulting in the loss of a frame in period.

Reply:
I see, so you call the display refresh timer manually on VSYNC event. 

I wonder if an event is really required for this. What about having a helper `lv_nuttx_refresh_display(disp)` which is called from `lv_nuttx_uv_vsync_poll_cb` and it calls `_lv_display_refr_timer()`. Or maybe `lv_nuttx_uv_vsync_poll_cb` could call `_lv_display_refr_timer()` directly.

I'm not using NuttX driver on daily basis, so probably you know it better, I'm just asking. :slightly_smiling_face:

Reply:
The VSYNC event is better when we consider multiple LVGL instances. The virtual displays(instances) rely on VSYNC signal to sync with physical display.

Reply:
> The VSYNC event is better when we consider multiple LVGL instances. 

Why? Calling `_lv_display_refr_timer(disp)` on vsync interrupt seems multi-instance independent.

Note that, if `lv_event_register_id` is used `LV_EVENT_VSYNC` needs to be added to `lv_globals` for multi instance support.

Reply:
The multiple LVGL instances will only have one connected to physical display. Thus adding a LVGL level VSYNC can easily forward this information to other instances.

Besides, the refresh timer should also be driven by VSYNC signal and we can unify this behavior.

Reply:
I see. Can we reuse `LV_EVENT_REFRESH` for this? [By definition](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_event.h#L79) this event is never sent by LVGL but it's for the user to indicate something requires a "refresh".

Reply:
`LV_EVENT_VSYNC` is only defined as an event for receiving hardware vblank signal, and it is not strongly bound to actual refresh or other tasks. It is recommended to keep the current precise definition.

Reply:
VSYNC is a standard event for GUI. It would be better not mixing it with `REFRESH`. The `refresh` should be an action driven by VSYNC.

For other GUI libs, I can find info about VSYNC too.
https://support.touchgfx.com/docs/basic-concepts/operating-system#touchgfx-os-wrappers
https://source.android.com/docs/core/graphics/implement-vsync

Reply:
In this case le's create a new event code for in in `lv_event.h`

Reply:
moved LV_EVENT_VSYNC to lv_event.h